changelog.d/13927.bugfix

```diff
@@ -0,0 +1 @@
+Fixes check_avatar_size_and_mime_type() to successfully update avatars on homeservers running on non-default ports which were mistakenly treated as remote homeserver while validating the avatar's size and mime type. Contributed by @ashfame.
```

Comment:
Imperative mood, please; and users don't care which function you changed, just what the bug was.

```suggestion
Fix a bug which prevented setting an avatar on homeservers which have an explicit port in their `server_name`. Contributed by @ashfame.
```

(Does it affect *all* homeservers with ports in their server name, or is this only a problem with certain configuration options?)

Reply:
> Does it affect all homeservers with ports in their server name

Yes!

Reply:
Right. I was under the impression that `check_avatar_size_and_mime_type` only did anything if you had specified a `max_avatar_size` and/or a `allowed_avatar_mimetypes`. No?

Reply:
Yes, my bad! You are right.

Reply:
ok well, can you clarify this in the changelog entry please.

Reply:
Done!